src/services/infra/InfraService.ts

```diff
@@ -49,27 +51,45 @@ export default class InfraService {
 
   private readonly queueService: InfraServiceProps["queueService"]
 
+  private readonly collaboratorsService: InfraServiceProps["collaboratorsService"]
+
   constructor({
     sitesService,
     reposService,
     deploymentsService,
     launchesService,
     queueService,
+    collaboratorsService,
   }: InfraServiceProps) {
     this.sitesService = sitesService
     this.reposService = reposService
     this.deploymentsService = deploymentsService
     this.launchesService = launchesService
     this.queueService = queueService
+    this.collaboratorsService = collaboratorsService
   }
 
-  createSite = async (creator: User, siteName: string, repoName: string) => {
+  createSite = async (
+    creator: User,
+    member: User,
+    siteName: string,
+    repoName: string
+  ) => {
     let site: Site | undefined // For error handling
+    const memberEmail = member.email
+    if (!memberEmail) {
+      logger.error(
+        `createSite: initial member for ${siteName} does not have associated email`
+      )
+      throw new Error(
+        `createSite: initial member for ${siteName} does not have associated email`
+      )
+    }
     try {
       // 1. Create a new site record in the Sites table
       const newSiteParams = {
         name: siteName,
-        apiTokenName: "", // TODO: figure this out
+        apiTokenName: "", // TODO: remove once DB has removed this param
```

Comment:
do we have an issue/jira for this?

Reply:
Added in https://github.com/isomerpages/isomercms-backend/pull/679/commits/926b0023857832cb6191e15e7932b39024ab3612!